app/src/main/java/ai/elimu/appstore/synchronization/AppListAdapter.java

```diff
@@ -195,7 +208,9 @@ public void onInstallComplete(@NonNull String packageName) {
                                     executorService.execute(new Runnable() {
                                         @Override
                                         public void run() {
-                                            apkFile.delete();
+                                            for (final File file : apkFiles) {
+                                                file.delete();
```

Comment:
To make testing easier, add some logging of which file is being deleted, e.g. by printing its filename.